### ISSUE TITLE
TMW trying to access protected method error

### ISSUE THREAD

**Issue by arieh** [ascott18/TellMeWhen#2124]
### WoW Version

Retail

### TellMeWhen Version

v10.2.0

### Describe the bug

Been seeing this error for a while:
```
6x [ADDON_ACTION_BLOCKED] AddOn 'TellMeWhen' tried to call the protected function 'resume()'.
[string "@!BugGrabber/BugGrabber.lua"]:481: in function <!BugGrabber/BugGrabber.lua:481>
[string "=[C]"]: in function `resume'
[string "@TellMeWhen/TellMeWhen.lua"]:2680: in function <TellMeWhen/TellMeWhen.lua:2669>
```

Don't think it causes any in-game issues. Unsure if it's related to a specific icon or just TMW itself. 

### Export Strings

```text
n/a
```

**Comment by ascott18** [comment]
Its a known issue - `resume` isn't actually protected. Are you able to reliably reproduce this?

https://github.com/Stanzilla/WoWUIBugs/issues/483

**Comment by arieh** [comment]
Every time I enter combat

**Comment by ascott18** [comment]
That would be great, or if you want to share your saved variables then I could dig in too

**Comment by arieh** [comment]
yeah ok after spending 5min on this I figure this is way above my head. It does seem on the surface that this should never happen since you're the one creating the coroutine in the first place...

Interesting tidbit is that it happens consistently after I reload when I hit a target dummy, but at some point later it stops happening. 

I'll share my saved variables, though I'm unsure how helpful it will be since my TMW config is huge :D

**Comment by arieh** [comment]
[TellMeWhen.zip](https://github.com/ascott18/TellMeWhen/files/13501061/TellMeWhen.zip)

**Comment by Grumsta** [comment]
I'm seeing the same error message, but only on my Frost Mage.

5x [ADDON_ACTION_BLOCKED] AddOn 'TellMeWhen' tried to call the protected function 'resume()'.
[string "@!BugGrabber/BugGrabber.lua"]:481: in function <!BugGrabber/BugGrabber.lua:481>
[string "=[C]"]: in function `resume'
[string "@TellMeWhen/TellMeWhen.lua"]:2680: in function <TellMeWhen/TellMeWhen.lua:2669>

**Comment by Alwies** [comment]
Found a way to replicate this (based on a tip from discord user Verazord). 

1. Have an item cooldown icon with "Range check" Checked (See below for import).
2. Enter combat (used training dummy room in Valdrakken)
3. Untarget your current target (escape).

Note: Only happens once after a /reload
Still doesn't seem to cause any issues other than the error message itself. 

Icon import
`^1^T^SShowTimer^B ^SType^Sitem ^SShowTimerText^B ^SName^S14 ^SStates^T ^N2^T ^SAlpha^N0.44 ^t^t^SRangeCheck^B ^SEnabled^B ^t^N102201^S~`~| ^Sicon^^`

**Comment by ascott18** [comment]
Coroutine usage has been obliterated in favor of our own custom state machine because it was causing so many problems. 9c4e885235896e286f6a33077848d1a183a13a4b  dddff42189e60a57774fd0d3d11b57aa155e8f7b
